### DISCUSSION TITLE
How to set Focus on TextInput in PopUp

### DISCUSSION THREAD

**Discussion by XiaoGen-Taiwan** [airsdk/Adobe-Runtime-Support#2148] [General]
Does anyone have an idea for setting the focus on a TextInput within a PopUp (SkinnablePopUpContainer) after that PopUp was opened.

I tried to set the focus by myTextInput.setFocus() within creationComplete but the TextInput looses the focus for whatever reason.
I would like to avoid using a timer.

**Comment by 2jfw** [comment]
I actually came across this just a few days ago and realized the focus (/FocusSkin) within popups for some components does not become visible without an explicit _second_ call (though tracing `stage.focus` does actually show correct focus component).

Try adding the additional line after `component.setFocus();`:
`component.focusManager.showFocus();`

**EDIT:** If this does not work, try calling both lines within a function in `callLater`.

**Reply by XiaoGen-Taiwan** [reply]
Hi @2jfw , thanks a lot for your reply. 
I couldn't figure out what was stealing the focus from the TextInput but 
setting the focus delayed (callLater) didn't work in my specific case.
Since I'm adding the PopUp's content dynamically, I wasn't able to reliably 
deal with the focusIn/focusOut Events within the content Group.

My solution: 
Step 1: PopUp (SkinnablePopUpContainer):
dispatching an openEvent down to the content container (a placeholder for the contentGroup) after the popUp is opened.  
Or, if using a showEffect dispatching an openEvent down to the content container at the effects end.

Step 2: Dynamic Content (Group):
listening to the PopUp's openEvent by adding a Listener when the content Group gets added to the PopUp.
Within that handler set the focus on the TextInput.
